app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class ArticlesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :validate_index_props, only: [:index]
+
+  def validate_index_props
+    @validator = ParamsValidator::ArticleSearchParamsValidator.new(params)
+    redirect_to root_path unless @validator.valid?
```

Comment:
This behavior is strange to me, maybe it is better to just ignore them?

Reply:
it's easier for me to validate all the data at the input and not before use

Reply:
Redirecting to root path is strange if there is any errors

Reply:
I clean query params - articles#index and root_path are same route